src/dmd/func.d

```diff
@@ -220,6 +236,7 @@ extern (C++) class FuncDeclaration : Declaration
     VarDeclaration vthis;               /// 'this' parameter (member and nested)
     VarDeclaration v_arguments;         /// '_arguments' parameter
     ObjcSelector* selector;             /// Objective-C method selector (member function only)
+    VarDeclaration selectorParameter;   /// Objective-C implicit selector parameter
```

Comment:
How is this different from HiddenParameters.selector?

Reply:
It's the same. `HiddenParameters` is just a way to return multiple values from `declareThis`. Since D doesn't have built-in way to return tuples and we're not allowed to use Phobos I added `HiddenParameters`. I named it `HiddenParameters.selector` and not `HiddenParameters.selectorParameter` because `HiddenParameters` already implies it's a parameter. But I can rename it if you prefer.

Reply:
Renamed `HiddenParameters.selector` to `HiddenParameters.selectorParameter`.